paddle/fluid/operators/recv_op.cc

```diff
@@ -36,26 +37,48 @@ class RecvOp : public framework::OperatorBase {
                const platform::Place& place) const override {
     auto outs = Outputs("Out");
     std::vector<std::string> epmap = Attr<std::vector<std::string>>("epmap");
+    std::vector<std::string> endpoints =
+        Attr<std::vector<std::string>>("endpoints");
+    auto client_var_name = Output("RPCClient");
+    PADDLE_ENFORCE_NOT_NULL(scope.FindVar(client_var_name),
+                            "Can not find variable '%s' in the scope.",
+                            client_var_name);
 
     platform::DeviceContextPool& pool = platform::DeviceContextPool::Instance();
     auto& ctx = *pool.Get(place);
 
+    platform::RecordEvent record_event(Type(), &ctx);
+    auto* client_var = scope.FindVar(client_var_name);
+    detail::RPCClient* rpc_client = client_var->GetMutable<detail::RPCClient>();
+
     for (size_t i = 0; i < outs.size(); i++) {
       VLOG(3) << "getting " << outs[i];
-      client_.AsyncGetVariable(epmap[i], ctx, scope, outs[i]);
+      rpc_client->AsyncGetVariable(epmap[i], ctx, scope, outs[i]);
     }
-    PADDLE_ENFORCE(client_.Wait());
-  }
+    PADDLE_ENFORCE(rpc_client->Wait());
 
- private:
-  mutable detail::RPCClient client_;
+    for (size_t i = 0; i < outs.size(); i++) {
+      VLOG(2) << "getting " << outs[i] << " from " << epmap[i];
+      rpc_client->AsyncGetVariable(epmap[i], ctx, scope, outs[i]);
```

Comment:
Why there are two `rpc_client->AsyncGetVariable(epmap[i], ctx, scope, outs[i]);` (another one at line 56) in this function?

Reply:
Sorry, it's a mistake, and have fixed it.